src/couch_replicator_httpc.erl

```diff
@@ -172,10 +181,14 @@ process_stream_response(ReqId, Worker, HttpDb, Params, Callback) ->
                 Ret = Callback(Ok, Headers, StreamDataFun),
                 Ret
             catch
+                throw:{maybe_retry_req, connection_closed} ->
+                    maybe_retry({connection_closed, mid_stream},
+                        Worker, HttpDb, Params);
                 throw:{maybe_retry_req, Err} ->
                     maybe_retry(Err, Worker, HttpDb, Params)
             end;
         R when R =:= 301 ; R =:= 302 ; R =:= 303 ->
+            backoff_success(HttpDb, Params),
```

Comment:
Yeah I counted as success for the purpose of rate limiting. I wasn't 100% sure about this but I erred on the optimistic side better to have more successes. This means for example requests would bump more against the limit more often.